types/mapcmd/put-all.go

```diff
@@ -0,0 +1,199 @@
+ * limitations under the License.
+ */
+package mapcmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/types"
+	"github.com/spf13/cobra"
+
+	hzcerror "github.com/hazelcast/hazelcast-commandline-client/errors"
+	"github.com/hazelcast/hazelcast-commandline-client/internal"
+	fds "github.com/hazelcast/hazelcast-commandline-client/types/flagdecorators"
+)
+
+func NewPutAll(config *hazelcast.Config) (*cobra.Command, error) {
+	var (
+		entries []types.Entry
+	)
+	var (
+		jsonEntryPath string
+		jsonEntries   map[string]interface{}
+	)
+	var (
+		mapName string
+		mapKeys,
+		mapValues,
+		mapValueTypes,
+		mapValueFiles []string
+	)
+	validateJsonEntryFlag := func() error {
+		if (len(mapKeys) |
+			len(mapValues) |
+			len(mapValueTypes) |
+			len(mapValueFiles)) != 0 {
+			return hzcerror.NewLoggableError(nil, fmt.Sprintf("%s is already set, there cannot be additional flags", fds.JsonEntryFlag))
+		}
+		return nil
+	}
+	validateValuesFlag := func() ([]byte, []int, error) {
+		valueCount := len(mapValues) + len(mapValueFiles)
+		if valueCount != len(mapKeys) {
+			return nil, nil, hzcerror.NewLoggableError(nil, "number of keys and values does not match")
+		}
+		vOrder, tOrder := ObtainOrderingOfValueFlags(os.Args)
+		if vOrder == nil {
+			return nil, nil, hzcerror.NewLoggableError(nil, "correct order of values cannot be taken")
+		}
+		return vOrder, tOrder, nil
+	}
+	executePutAll := func(ctx context.Context, cmd *cobra.Command, m *hazelcast.Map, entries []types.Entry) error {
+		var err error
+		err = m.PutAll(ctx, entries...)
+		if err != nil {
+			cmd.Println("Cannot put given entries")
+			isCloudCluster := config.Cluster.Cloud.Enabled
+			if networkErrMsg, handled := internal.TranslateNetworkError(err, isCloudCluster); handled {
+				err = hzcerror.NewLoggableError(err, networkErrMsg)
+			}
+			return err
+		}
+		return nil
+	}
+	cmd := &cobra.Command{
+		Use:   "put-all --name mapname [--key keyname]... [[--value-file file | --value value][--value-type type]]...",
+		Short: "Put values to map",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx, cancel := context.WithTimeout(cmd.Context(), time.Second*3)
+			defer cancel()
+			if jsonEntryPath != "" {
+				if err := validateJsonEntryFlag(); err != nil {
+					return err
+				}
+				data, err := ioutil.ReadFile(jsonEntryPath)
+				if err != nil {
+					return err
+				}
+				if err = json.Unmarshal(data, &jsonEntries); err != nil {
+					return hzcerror.NewLoggableError(err, "given json map entry file is in invalid format")
+				}
+				for key, jsv := range jsonEntries {
+					switch jsv.(type) {
+					case nil:
+						// ignore null json values
+					case string, float64, bool, []interface{}:
+						entries = append(entries, types.Entry{Key: key, Value: jsv})
+					case map[string]interface{}:
+						var nJson interface{}
+						mj, _ := json.Marshal(jsv)
```

Comment:
Why do we marshal it?

Reply:
I marshaled it to be applied client json serialization support. I have a value as `map[string]interface{}` then I need to convert into `[]byte` to normalize and return as `interface{}`.

Reply:
Makes sense thanks